discovery_engine_core/ai/ai/src/kps/config.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::{cmp::Ordering, time::Duration};
+
+use displaydoc::Display;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crate::utils::{nan_safe_f32_cmp_desc, serde_duration_as_days, SECONDS_PER_DAY_U64};
+
+/// Configurations of the kps system.
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[must_use]
+pub struct Config {
+    #[serde(with = "serde_duration_as_days")]
+    horizon: Duration,
```

Comment:
❓ Config for Coi also has this `horizon` setting, won't it get confusing which one to use? Or maybe we need to have this configured separately?

Reply:
good point, the `horizon` is used in the `coi::context` as well as `kps::key_phrase`. i removed it from the `KpsConfig` such that there is a single source of truth in the `CoiConfig`. as the `KpsSystem` requires the `CoiSystem` anyways, it can also access its config and doesn't need to duplicate the value.